flow/core/kernel/scenario/base.py

```diff
@@ -301,17 +428,111 @@ def gen_custom_start_pos(self, initial_config, num_vehicles, **kwargs):
             see flow/core/params.py
         num_vehicles : int
             number of vehicles to be placed on the network
-        kwargs : dict
-            extra components, usually defined during reset to overwrite initial
-            config parameters
 
         Returns
         -------
         list of tuple (float, float)
             list of start positions [(edge0, pos0), (edge1, pos1), ...]
         list of int
             list of start lanes
-        list of float
-            list of start speeds
         """
-        raise NotImplementedError
+        return self.network.gen_custom_start_pos(
+            cls=self,
+            initial_config=initial_config,
+            num_vehicles=num_vehicles,
+        )
+
+    def _get_start_pos_util(self, initial_config, num_vehicles):
+        """Prepare initial_config data for starting position methods.
+
+        Performs some pre-processing to the initial_config and **kwargs terms,
+        and returns the necessary values for all starting position generating
+        functions.
+
+        Parameters
+        ----------
+        initial_config : InitialConfig type
+            see flow/core/params.py
+        num_vehicles : int
+            number of vehicles to be placed on the network
+
+        Returns
+        -------
+        x0 : float
+            starting position of the first vehicle, in meters
+        min_gap : float
+            minimum gap between vehicles
+        bunching : float
```

Comment:
Nit: I think we should adjust this comment, I still don't understand _exactly_ how bunching works.

Reply:
We need to document all these things (and maybe delete bunching tbh). There is an issue about this